### ISSUE TITLE
(web / browser) Secure Note will not accept 9275 length note

### ISSUE THREAD

**Issue by troyengel** [bitwarden/web#172]
This happens in both the Vault on the web and in the browser extension - I am attempting to store a GPG key which has been exported in ascii-armored format, the entire length of content is 9,275 characters / bytes (verified length with `wc -m`, `wc -c`, and vim on the saved file). The error is "Secure Note has maximum string of 10,000 characters":

![screenshot at 2018-02-02 19-18-12](https://user-images.githubusercontent.com/402696/35761431-f37faff2-084d-11e8-95fb-a4e3193dbb1e.png)

This is the standard export format and is basically impossible for me to alter and still be usable, i.e.:
```
-----BEGIN PGP PRIVATE KEY BLOCK-----

1234567890...
abcdefghijkl...
...
-----END PGP PRIVATE KEY BLOCK-----
```
I am below the threshold listed (what luck - I don't get to choose how big this is :) ) but get the error, making me believe it's coming from the server (?) not the apps - I'm using the hosted Premium service not my own servers. As I'm not sure if this is "core" or not I'll file here...?  thanks!

**Comment by troyengel** [comment]
FYI: this is a RSA 2048 key with a RSA 2048 subkey (basically, the GnuPG defaults) with three associated signing email identities, nothing crazy or out of the ordinary as far as a GPG key goes. The export is a standard textbook format, once again pretty much defaults of GnuPG `--armor` output.

**Comment by kspearrin** [comment]
@troyengel The actual limit is lower since the server error you are seeing in on the encrypted value of the string you are putting in the notes field. Probably somewhere around 5000.

You should store these types of things as binary file attachments rather than in the notes field.

**Comment by troyengel** [comment]
@kspearrin The explanation makes perfect sense, thank you - perhaps update the error message received to be a bit more descriptive? The way it reads is how I got here, completely missed that subtle difference. But now that you mention it, completely on board.

I've attached the items to Notes, but attachments are not exported with your passwords (to CSV) which means they're "locked" on the remote server, not a part of my password data dump backup (which ironically I GPG encrypt) - not portable. I really would like it if we could increase the text field size to be able to store a GPG key such as this so we don't have to use attachments... :)

Feature request to increase storeable (post-encryption) Note length a bit higher?
